src/database/migrations/0028-alter-field_type-enum.sql

```diff
@@ -0,0 +1,9 @@
+CREATE TYPE updated_field_type AS ENUM ('string', 'number', 'phone_number', 'email', 'url', 'boolean');
+ALTER TABLE base_fields
+  ALTER COLUMN data_type TYPE updated_field_type USING
+  CASE
+      WHEN data_type = 'URL' THEN 'url'::updated_field_type
+      ELSE data_type::text::updated_field_type
+  END;
+DROP TYPE field_type CASCADE;
+ALTER TYPE updated_field_type RENAME to field_type;
```

Comment:
This is great, and I have no qualms.  I was curious though so asked GPT how it would do it:

```
-- Step 1: Add the new value to the enum type
ALTER TYPE user_type ADD VALUE 'boss';

-- Step 2: Update the table to use the new value
UPDATE users SET user_type = 'boss' WHERE user_type = 'BOSS';

-- Step 3: Remove the old value (PostgreSQL does not support removing enum values directly)
-- So instead, create a new type without 'BOSS' and replace the old type

-- Create a new enum type without 'BOSS'
CREATE TYPE user_type_new AS ENUM('boss', /* other values */);

-- Alter the table to use the new enum type
ALTER TABLE users
  ALTER COLUMN user_type TYPE user_type_new
  USING user_type::text::user_type_new;

-- Drop the old enum type
DROP TYPE user_type;

-- Rename the new enum type to the original name
ALTER TYPE user_type_new RENAME TO user_type;
```

It's very similar to your approach, but by adding the `url` value to the original type / updating the fields to use `url`, and then swapping the enum to one without the (now unused `URL`) you could avoid the complexity of the `CASE` statement.

Reply:
@slifty I think that avoiding a case statement is preferable!

Reply:
Following up here since Harry and I went offline to do some rapid back and forth -- Harry did implement this change but psql doesn't like using enums that aren't committed in a transaction (idk why, but whatever.)

Having two migrations felt nastier than a case statement, so he landed back where he started with the case statement!

That was fun :D